RNCryptor/RNDecryptor.m

```diff
@@ -198,13 +201,19 @@ - (void)addData:(NSData *)theData
     NSUInteger HMACLength = self.HMACLength;
     if (self.inData.length > HMACLength) {
       NSData *data = [self.inData _RNConsumeToIndex:self.inData.length - HMACLength];
-      [self decryptData:data];
+      if (data) {
+        [self decryptData:data];
+      }
     }
   }
 }
 
 - (BOOL)updateOptionsForPreamble:(NSData *)preamble
 {
+  if (!preamble) {
+    return NO;
+  }
+    
```

Comment:
Looks like this is never called with nil, though it would crash if it was.  The preamble length is checked in the calling method, with different error handling.